lib/levelup.js

```diff
@@ -62,14 +61,17 @@ function LevelUP (location, options, callback) {
     options = {}
   }
 
-  if ((!options || typeof options.db !== 'function') && typeof location !== 'string') {
-    error = new InitializationError(
-        'Must provide a location for the database')
-    if (callback) {
-      return process.nextTick(function () {
-        callback(error)
-      })
-    }
+  if (!options || typeof options.db !== 'function') {
```

Comment:
@juliangruber If we're happy with this PR we might want to consider squashing this commit with the next. It was easier for me to follow along what you did on `v2` branch before removing `location` completely.